File: ThreadLab2C#/MinimumFinder/ArrayMinimumManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    public class ArrayMinimumManager
    {
        private int _min = int.MaxValue;
        private readonly object _lock = new object();

        public void UpdateMin(int newMin)
        {
            lock (_lock)
            {
                if (newMin < _min)
                {
                    _min = newMin;
                }
            }
        }

        public int GetMin()
        {
            lock (_lock)
            {
                return _min;
            }
        }
    }
}

File: ThreadLab2Java/Threads/src/ArrayMinimumManager.java
import java.util.concurrent.locks.ReentrantLock;

public class ArrayMinimumManager {
    private int min = Integer.MAX_VALUE;
    private final ReentrantLock lock = new ReentrantLock();

    public void updateMin(int newMin) {
        lock.lock();
        try {
            if (newMin < min) {
                min = newMin;
            }
        } finally {
            lock.unlock();
        }
    }

    public int getMin() {
        lock.lock();
        try {
            return min;
        } finally {
            lock.unlock();
        }
    }
}